marlowe-dashboard-client/src/Play/State.purs

```diff
@@ -236,11 +236,15 @@ handleAction input@{ currentSlot } AdvanceTimedoutSteps = do
     $ over
         (_contractsState <<< _contracts <<< traversed <<< filtered (\contract -> contract.executionState.mNextTimeout /= Nothing && contract.executionState.mNextTimeout <= Just currentSlot))
         (applyTimeout currentSlot)
-  -- if the modification changed the currently selected step, that means the card for the contract
-  -- that was changed is currently open, so we need to realign the step cards
+  -- If the modification changed the currently selected step, that means the card for the contract
+  -- that was changed is currently open, so we need to realign the step cards. We also call the
+  -- CancelConfirmation action - because if the user had the action confirmation card open for an
+  -- action in the current step, we want to close it (otherwise they could confirm an action that
+  -- is no longer possible).
   selectedStep' <- peruse $ _selectedContract <<< _selectedStep
-  when (selectedStep /= selectedStep')
-    $ for_ selectedStep' (handleAction input <<< ContractAction <<< Contract.MoveToStep)
+  when (selectedStep /= selectedStep') do
+    for_ selectedStep' (handleAction input <<< ContractAction <<< Contract.MoveToStep)
+    handleAction input $ ContractAction Contract.CancelConfirmation
```

Comment:
I agree that we should close the cancel confirmation card on timeout, but we should also fix this TODO as part of this commit, so we don't accidently close the wrong card

```purescript
-- line 301
Contract.CancelConfirmation -> handleAction input CloseCard -- TODO: guard against closing the wrong card
```